week2/index.js

```diff
@@ -0,0 +1,135 @@
+
+const fs = require('fs');
+const readline = require('readline');
+let todos = [];
+
+
+if (process.argv[2] !== undefined) {
+    start();
+} else {
+    console.log('Please specify an option to use, here is the manual\n');
+    manualReader('./help.txt');
+}
+
+
+function start() {
+    switch (process.argv[2].trim().toLowerCase()) {
+        case 'help':
+            manualReader('./help.txt');
+            break;
+        case 'add':
+            checkFileContent().then((item) => addTodoItem(item)).catch((err) => console.log(err));
+            break;
+        case 'remove':
+            itemRemover(process.argv[3]);
+            break;
+        case 'list':
+            console.log('The items on your todo list: ');
+            todoFileReader('./todos.txt');
+            break;
+        case 'reset':
+            itemsResetter();
+            break;
+        default:
+            console.log('No such option! Take a look at the manual\n');
+            manualReader('./help.txt');
+            break;
+    }
+}
+
+
+function manualReader(fileName) {
+    fs.readFile(fileName, 'UTF-8', (err, data) => {
+        console.log(data);
+    });
+}
+
+//  display the todo file once called
+function todoFileReader(fileName) {
+    fs.exists(fileName, (exists) => {
+        //see if file exists to add the existing items to the array
+        if (exists) {
+            fs.readFile(fileName, 'UTF-8', (err, data) => {
+                if (data === "") {
+                    console.log('There are no items yet, try using \'node index.js add "some todo item"\'');
+                    return;
+                }
+                data = data.split('\r\n');
+                let text = data.reduce(((last, item, index) => last += `${index + 1}. ${item}\n`), ``);
+                console.log(text);
+            });
+        } else {
+            console.log('There are no items yet, try using \'node index.js add "some todo item"\'');
+        }
+    });
+}
+
+
+
+//check if there's already a file and to push the item in
+
+function checkFileContent() {
+    return new Promise(function (resolve, reject) {
+        fs.exists('./todos.txt', (exists) => {
+            if (!exists) {
+                resolve(process.argv[3]);
+                return;
+            }
+            fs.readFile('todos.txt', 'UTF-8', (err, data) => {
+                todos = data.trim().split('\r\n');
+                resolve(process.argv[3]);
+            });
+        });
+    });
+}
+
+//add the new item to the array
+function addTodoItem(todoItem) {
+    if (todoItem !== '') {
+        if (todoItem.trim() !== '') {
+            todos.push(todoItem.trim());
+            todosWriter(todos);
+        }
+    } else {
+        console.log('Not the correct usage! Try "node index.js help"');
+    }
+}
+
+// rewrite existing file with new todos array 
+function todosWriter(itemsArr) {
+    itemsArr.filter((a) => a !== '');
```

Comment:
Is there a particular case you're hoping to catch here? Or maybe a bug you encountered that this fixes?
First of all, you check for empty strings earlier before you add items, so we're already pretty sure there won't be an empty string here.
But the other problem is that `Array.prototype.filter()` returns the new array, it doesn't modify the original. Since you don't save the output of this `filter()` call, you aren't actually using the `filter()` at all.